src/isort/mod.rs

```diff
@@ -585,29 +586,49 @@ pub fn format_imports(
 
         let mut is_first_statement = true;
 
-        // Format `StmtKind::Import` statements.
-        for (alias, comments) in &import_block.import {
-            output.append(&format::format_import(
-                alias,
-                comments,
-                is_first_statement,
-                stylist,
-            ));
-            is_first_statement = false;
-        }
-
-        // Format `StmtKind::ImportFrom` statements.
-        for (import_from, comments, trailing_comma, aliases) in &import_block.import_from {
-            output.append(&format::format_import_from(
-                import_from,
-                comments,
-                aliases,
-                line_length,
-                stylist,
-                force_wrap_aliases,
-                is_first_statement,
-                split_on_trailing_comma && matches!(trailing_comma, TrailingComma::Present),
-            ));
+        let it = import_block
+            .import
+            .into_iter()
+            .map(AnyImport::Import)
+            .chain(
+                import_block
+                    .import_from
+                    .into_iter()
+                    .map(AnyImport::ImportFrom),
+            )
+            .sorted_by(|a, b| {
+                if force_sort_within_sections {
+                    cmp_any_import(a, b)
```

Comment:
Maybe instead, `let mut it`, and then call `sorted_by` on it if `force_sort_within_sections`? That way, we avoid the oddity of `else { Ordering::Greater }`.

Reply:
I was sort of able to do this, although I had to `.collect` before (and within) the `if` block to ensure `it ` was the same type.

Reply:
If we use `sort_by` instead of `sorted_by`, we can avoid that allocation.